src/haz3lcore/lang/MakeTerm.re

```diff
@@ -243,14 +243,20 @@ and exp_term: unsorted => (Exp.term, list(Id.t)) = {
         | term => ret(ListLit([term]))
         }
       | (["test", "end"], [Exp(test)]) => ret(Test(test))
+      | (["hint", "test", "end"], [Exp(hint), Exp(test)]) =>
```

Comment:
this seems to indicate that the parsing of a hint form fails entirely if the hint is not a string? seems like this could have negative consequences in-editor. i can't test it though as hints don't seem to work in-editor? tried to enter "hint" and it doesn't expand